internal/azuremodels/azure_client.go

```diff
@@ -218,28 +208,34 @@ func (c *AzureClient) ListModels(ctx context.Context) ([]*ModelSummary, error) {
 	decoder := json.NewDecoder(resp.Body)
 	decoder.UseNumber()
 
-	var searchResponse modelCatalogSearchResponse
-	err = decoder.Decode(&searchResponse)
+	var catalog githubModelCatalogResponse
+	err = decoder.Decode(&catalog)
 	if err != nil {
 		return nil, err
 	}
 
-	models := make([]*ModelSummary, 0, len(searchResponse.Summaries))
-	for _, summary := range searchResponse.Summaries {
+	models := make([]*ModelSummary, 0, len(catalog))
+	for _, catalogModel := range catalog {
+		// Determine task from supported modalities - if it supports text input/output, it's likely a chat model
 		inferenceTask := ""
-		if len(summary.InferenceTasks) > 0 {
-			inferenceTask = summary.InferenceTasks[0]
+		if slices.Contains(catalogModel.SupportedInputModalities, "text") && slices.Contains(catalogModel.SupportedOutputModalities, "text") {
+			inferenceTask = "chat-completion"
+		}
+
+		modelKey, err := modelkey.ParseModelKey(catalogModel.ID)
+		if err != nil {
+			return nil, fmt.Errorf("parsing model key %q: %w", catalogModel.ID, err)
```

Comment:
Ahh you calling it a "model key" here when printing out the `ID` makes me think that we will indeed have the publisher + model name combined in the field, so probably don't need such a test as I suggested above. 👍🏻